gsuite/resource_user.go

```diff
@@ -448,10 +448,12 @@ func resourceUserCreate(d *schema.ResourceData, meta interface{}) error {
 		}, config.TimeoutMinutes)
 
 		var locatedUser *directory.User
-		for _, existingUser := range existingUsers.Users {
-			if existingUser.PrimaryEmail == user.PrimaryEmail {
-				locatedUser = existingUser
-				break
+		if existingUsers.Users != nil {
```

Comment:
It should be fine to add this check but I'm not sure if it's needed since ranging over a `nil` slice should be fine as the below code demonstrates.

```
package main

import (
	"fmt"
)

func main() {

	var slice []string

        slice = nil

	for index, item := range slice {
		fmt.Printf("%d: %s\n", index, item)
	}
}

```

Am I missing something?

Reply:
Here's a repro:
```
package main

import (
        "fmt"
        directory "google.golang.org/api/admin/directory/v1"
)

func main() {
        fmt.Println("Hello, playground")
        var existingUsers *directory.Users

        for index, item := range existingUsers.Users {
                fmt.Printf("%d: %s\n", index, item)
        }
}
```

```
$ go run test.go
Hello, playground
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x660ef9]

goroutine 1 [running]:
main.main()
        /home/chase/Sites/blockstream/devops/kubernetes/test.go:12 +0x89
exit status 2
```

Reply:
The check should probably be for if the `existingUsers` is nil not `existingUsers.Users`.

```
package main

import (
	"fmt"
)

type User struct {
}

type Users struct {
	Users []*User
}

func main() {
	var goodUsers []*User
	goodUsers = nil
	for index, item := range goodUsers {
		fmt.Printf("good %d: %s\n", index, item)
	}

	nilUsers := &Users{}
	nilUsers.Users = nil
	for index, item := range nilUsers.Users {
		fmt.Printf("nil %d: %s\n", index, item)
	}

	var existingUsers *Users
	existingUsers = nil
	for index, item := range existingUsers.Users {
		fmt.Printf("bad %d: %s\n", index, item)
	}
}
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1094821]

goroutine 1 [running]:
main.main()
	main.go:29 +0x31
exit status 2
```

Reply:
Good catch, since this is breaking a pushed a fix to master and cut a new release https://github.com/DeviaVir/terraform-provider-gsuite/releases/tag/v0.1.39